fedimint-client/src/module/mod.rs

```diff
@@ -102,6 +102,87 @@ pub trait ClientModule: Debug + MaybeSend + MaybeSync + 'static {
     ) -> anyhow::Result<()> {
         anyhow::bail!("Wiping not supported");
     }
+
+    /// Does this module support being a primary module
+    ///
+    /// If it does it must implement:
+    ///
+    /// * [`Self::create_sufficient_input`]
+    /// * [`Self::create_exact_output`]
+    /// * [`Self::await_primary_module_output`]
+    /// * [`Self::get_balance`]
+    /// * [`Self::subscribe_balance_changes`]
+    fn supports_being_primary(&self) -> bool {
+        false
+    }
+
+    // TODO: unclear if we should return a vec of inputs
+    /// Creates an input of **at least** a given `min_amount` from the holdings
+    /// managed by the module.
+    ///
+    /// If successful it returns:
+    /// * A set of private keys belonging to the input for signing the
+    ///   transaction
+    /// * The input of **at least** `min_amount`, the actual amount might be
+    ///   larger, the caller has to handle this case and possibly generate
+    ///   change using `create_change_output`.
+    /// * A closure that generates states belonging to the input. This closure
+    ///   takes the transaction id of the transaction in which the input was
+    ///   used and the input index as input since these cannot be known at time
+    ///   of calling `create_funding_input` and have to be injected later.
+    ///
```

Comment:
nit unrelated to this PR: We might want to move that documentation to `ClientInput`. Originally these items were returned as a tuple, that's why the documentation is still here.